yb_migrate/src/utils/commonVariables.go

```diff
@@ -94,3 +100,41 @@ type Issue struct {
 	Suggestion   string `json:"suggestion"`
 	GH           string `json:"GH"`
 }
+
+type DataFileDescriptor struct {
+	FileType      string           `json:"FileType"`
+	TableRowCount map[string]int64 `json:"TableRowCount"`
+	Delimiter     string           `json:"Delimiter"`
+	HasHeader     bool             `json:"HasHeader"`
+}
+
+func (dfd *DataFileDescriptor) StoreDataFileDescriptor(exportDir string) {
+	dfdFilePath := exportDir + "/metainfo/flags/dataFileDescriptor"
+	log.Infof("storing DataFileDescriptor at %q")
+
+	bytes, err := json.MarshalIndent(dfd, "", "\t")
+	if err != nil {
+		ErrExit("Marshalling the dfd struct: %v", err)
+	}
+
+	err = ioutil.WriteFile(dfdFilePath, bytes, 0644)
+	if err != nil {
+		ErrExit("writing DataFileDescriptor at %q: %v", dfdFilePath, err)
+	}
+}
+
+func (dfd *DataFileDescriptor) LoadDataFileDescriptor(exportDir string) {
```

Comment:
This function needs a change. Define a function with following signature in the `datafile` package:

```
func OpenDescriptor(exportDir) *Descriptor
```

Reply:
done